source/iot/Mqtt5Client.cpp

```diff
@@ -729,7 +730,17 @@ namespace Aws
 
         Mqtt5CustomAuthConfig &Aws::Iot::Mqtt5CustomAuthConfig::WithTokenSignature(Crt::String tokenSignature)
         {
-            m_tokenSignature = std::move(tokenSignature);
+            if (tokenSignature.find('%') != tokenSignature.npos)
+            {
+                // We can assume that a base 64 value that contains a '%' character has already been uri encoded
+                m_tokenSignature = tokenSignature;
+            }
+            else
+            {
+                m_tokenSignature =
+                    Aws::Crt::Io::EncodeQueryParameterValue(aws_byte_cursor_from_c_str(tokenSignature.c_str()));
```

Comment:
Debatable nit: if change EncodeQueryParameterValue to accept `const String&`, the calling code will be less verbose

Reply:
From a utility perspective, taking a byte cursor is quite a bit more useful.